### PR TITLE
Lecture 6 HW is done

### PR DESCRIPTION
**Pull request by Kornyshev**
image

### CODE REVIEW

File: lecture6/hw_ig.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                for item in merge_elems(*elem):
+                    yield item
+
+
+def map_like(fun, *elems):
+    for item in elems:
+        try:
+            yield fun(item)
+        except (TypeError, IndexError) as e:
+            yield f"{item}: '{type(item).__name__}' object {str(e).lower()}"
```
Comment (AnastasiiaGor) [inline lecture6/hw_ig.py]: f"{item}: {e}" is enough to print required string (str() is called implicitly)

File: lecture6/hw_ig.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+def merge_elems(*elems):
+    for elem in elems:
+        if not hasattr(elem, '__iter__'):
+            yield elem
+        else:
+            if isinstance(elem, str) and len(elem) == 1:
+                yield elem
+            else:
+                for item in merge_elems(*elem):
```
Comment (AnastasiiaGor) [inline lecture6/hw_ig.py]: 'yield from' might be used here but it's not essential :)

**Review by AnastasiiaGor — APPROVED** [review]

File: lecture6/hw_ig.py (hunk 1 above)
Comment (Kornyshev) [reply lecture6/hw_ig.py]: Done.

File: lecture6/hw_ig.py (hunk 2 above)
Comment (Kornyshev) [reply lecture6/hw_ig.py]: Done.

**Pull request merged**